src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/gerritnotifier/model/BuildMemoryReportContainerTest.java

```diff
@@ -0,0 +1,111 @@
+package com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.model;
+
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.ChangeBasedEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.GerritTriggeredEvent;
+import com.sonymobile.tools.gerrit.gerritevents.dto.events.PatchsetCreated;
+import hudson.model.AbstractProject;
+import hudson.model.ItemGroup;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests {@link BuildMemory} internal container.
+ * The main requirement is not merging non-equal events with same hash code.
+ *
+ * @author Robert Sandell &lt;rsandell@cloudbees.com&gt;.
+ */
+@RunWith(Parameterized.class)
+public class BuildMemoryReportContainerTest {
+
+    private final GerritTriggeredEvent o1;
+    private final GerritTriggeredEvent o2;
+    private final int expected;
+    private final BuildMemory memory;
+
+    static class PatchsetCreatedWithPredefinedHash extends PatchsetCreated {
+        private final int hashCode;
+        private final UUID uuid;
+
+        private PatchsetCreatedWithPredefinedHash(int hashCode) {
+            this.hashCode = hashCode;
+            this.uuid = UUID.randomUUID();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return o instanceof PatchsetCreatedWithPredefinedHash &&
+                    uuid.equals(((PatchsetCreatedWithPredefinedHash) o).uuid);
+
+        }
+
+        @Override
+        public int hashCode() {
+            return hashCode;
+        }
+
+        @Override
+        public String toString() {
+            return "PatchsetCreatedWithPredefinedHash{" +
+                    "uuid=" + uuid +
+                    '}';
+        }
+    }
+
+    /**
+     * The different scenarios.
+     *
+     * @return a list of parameters to the constructor.
+     * @see Parameterized
+     */
+    @Parameterized.Parameters(name = "{index}: {0}, {1} == {2}")
+    public static Iterable<Object[]> permutations() {
+        ChangeBasedEvent event1 = new PatchsetCreatedWithPredefinedHash(777);
+        ChangeBasedEvent event2 = new PatchsetCreatedWithPredefinedHash(777);
+        return Arrays.asList(new Object[][]{
+                {null, null, 1},
+                {null, event1, 2},
+                {event1, null, 2},
+                {event1, event1, 1},
+                {event2, event2, 1},
+                {event1, event2, 2},
```

Comment:
Should be one, if they have the same hashcode, they essentially are the same event even though it is different object instances.
An event might be just one instance when a series of builds are triggered, but when those builds are restored from disk the events in those builds will be different instances, and the quick way to see that two instances respresents the same event is to use the `hashCode`.